contracts/src/blackjack.cpp

```diff
@@ -297,15 +299,19 @@ void blackjack::on_random(uint64_t ses_id, checksum256 rand) {
             if (res == outcome::dealer || card_game::get_weight(state_itr->active_cards) == 21) {
                 if (!state_itr->has_split() || state_itr->second_round) {
                     auto [win, dealer_cards] = compare_and_finish(state_itr, ante, rand, std::move(deck));
-                    end_game(get_session(ses_id).deposit + win, std::move(dealer_cards), {player_card});
+                    end_game(ses_id, win, std::move(dealer_cards), {player_card});
                     return;
                 } else {
                     finish_first_round(state_itr);
                 }
             }
-            send_game_message(std::vector<param_t>{player_card.get_value()});
-            update_state(state_itr, game_state::require_play);
             require_action(action::play);
+            update_state(state_itr, game_state::require_play);
+            if (!state_itr->has_split() && state_itr->active_cards.size() == 2) {
```

Comment:
why `active_cards.size() == 2` ? oh, I see, here we have old `state_itr` (hasn't updated). Little bit implicit logic...

Reply:
i agree
any ideas how to improve it?

Reply:
e.g return new state from `update_state` method